### PR TITLE
Improvements for SearchParamManager

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
* Removes the CosmosDbStatusRegistryInitializer and moves the initialization logic into SearchParamManager so it can be shared with the SQL implementation
* Uses pre-computed support defined in FilebasedSearchParameterRegistry to speed up IsSupported checks

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs [hunk 1]
```diff
@@ -22,82 +25,143 @@ public class SearchParameterStatusManager : IRequireInitializationOnFirstRequest
...
+                        else
+                        {
+                            // Registry converter information is different, re-check parameter support.
+                            p.IsSupported = _searchParameterSupportResolver.IsSearchParameterSupported(p);
+                        }
 
-                    // Check if this parameter is now supported.
-                    p.IsSupported = _searchParameterSupportResolver.IsSearchParameterSupported(p);
+                        p.IsSearchable = p.IsSupported && isNewDatabase;
+                        updated.Add(p);
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs:145]: For my understanding, why isn't there a check similar to the one on lines 118-120 before adding `p` to `updated` here?

File: src/Microsoft.Health.Fhir.Core/Features/Search/ISearchIndexer.cs [hunk 2]
```diff
@@ -20,5 +20,11 @@ public interface ISearchIndexer
         /// <param name="resource">The resource to extract the search indices from.</param>
         /// <returns>An <see cref="IEnumerable{SearchIndex}"/> that contains the search index entries.</returns>
         IReadOnlyCollection<SearchIndexEntry> Extract(ResourceElement resource);
+
+        /// <summary>
+        /// Computers a hash of the currently supported FHIR to SearchValue types
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.Core/Features/Search/ISearchIndexer.cs:25]: Nit: "Computes"

File: src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Converters/IFhirElementToSearchValueTypeConverterManager.cs [hunk 3]
```diff
@@ -20,5 +20,11 @@ public interface IFhirElementToSearchValueTypeConverterManager
         /// <param name="converter">When this method returns, contains the converter associated with the FHIR element type if the FHIR element type exists; otherwise, <c>null</c>.</param>
         /// <returns><c>true</c> if the converter exists; otherwise, <c>false</c>.</returns>
         bool TryGetConverter(Type fhirElementType, Type searchValueType, out IFhirElementToSearchValueTypeConverter converter);
+
+        /// <summary>
+        /// Computers a hash of the currently supported FHIR to SearchValue types
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Converters/IFhirElementToSearchValueTypeConverterManager.cs:25]: Same typo as above

File: src/Microsoft.Health.Fhir.R5.Core/Features/Definition/unsupported-search-parameters.json [hunk 4]
```diff
@@ -93,19 +87,6 @@
...
-        "http://hl7.org/fhir/SearchParameter/ResearchDefinition-predecessor",
-        "http://hl7.org/fhir/SearchParameter/ResearchDefinition-successor",
-        "http://hl7.org/fhir/SearchParameter/ResearchDefinition-version",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-composed-of",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-depends-on",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-derived-from",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-predecessor",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-successor",
-        "http://hl7.org/fhir/SearchParameter/ResearchElementDefinition-version",
-        "http://hl7.org/fhir/SearchParameter/RiskEvidenceSynthesis-version",
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.R5.Core/Features/Definition/unsupported-search-parameters.json:108]: Are these now supported?

File: src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs (hunk 1 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs:145]: The above check only adds it if it has changed, this one is always new

File: src/Microsoft.Health.Fhir.R5.Core/Features/Definition/unsupported-search-parameters.json (hunk 4 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.R5.Core/Features/Definition/unsupported-search-parameters.json:108]: Need to double check this when I get a chance, this changed when switching to `BaseFhirParser.Inspector`. Possibly it was able to resolve more types

File: src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Converters/FhirElementToSearchValueTypeConverterManager.cs [hunk 5]
```diff
@@ -38,5 +40,10 @@ public bool TryGetConverter(Type fhirElementType, Type searchValueType, out IFhi
 
             return _converterDictionary.TryGetValue((fhirElementType, searchValueType), out converter);
         }
+
+        public string ComputeHash()
+        {
+            return JsonConvert.SerializeObject(_converterDictionary.Keys).ComputeHash();
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Converters/FhirElementToSearchValueTypeConverterManager.cs:46]: Order these to make the hash more consistent

**Comment by brendankowitz** [comment]
Closing this for now

**Pull request closed without merging**